Tools/build/deepfreeze.py

```diff
@@ -491,7 +491,8 @@ def generate(args: list[str], output: TextIO) -> None:
 parser = argparse.ArgumentParser()
 parser.add_argument("-o", "--output", help="Defaults to deepfreeze.c", default="deepfreeze.c")
 parser.add_argument("-v", "--verbose", action="store_true", help="Print diagnostics")
-parser.add_argument('args', nargs="+", help="Input file and module name (required) in file:modname format")
+parser.add_argument("-f", "--file", help="reads rule lines from a file")
+parser.add_argument('args', nargs="*", help="Input file and module name (required) in file:modname format")
```

Comment:
This makes the parser check that you've specified either args or --file
```suggestion
group = parser.add_mutually_exclusive_group(required=True)
group.add_argument("-f", "--file", help="reads rule lines from a file")
group.add_argument('args', nargs="*", default=(),
                   help="Input file and module name (required) in file:modname format")
```